src/main/java/org/mule/tools/apikit/output/MuleConfigGenerator.java

```diff
@@ -64,112 +71,249 @@ public MuleConfigGenerator(List<ApikitMainFlowContainer> apis, List<GenerationMo
+   * the apikit configuration.
+   * @param configRef apikit configuration reference
+   * @param configuration file from the app - could contain apikit configuration
+   * @return true/false depending on existence of any file matching the configuration.
+   */
+  private boolean filterConfigurations(String configRef, MuleConfig configuration) {
+    Stream<APIKitConfig> apikitConfigurationStream = configuration.getApikitConfigs().stream();
+    List<APIKitConfig> filteredApikitConfigurations = apikitConfigurationStream
+        .filter(apikitConfig -> apikitConfig.getName().contains(configRef)).collect(Collectors.toList());
+    return filteredApikitConfigurations.size() > 0;
+  }
 
+  /**
+   * Looks for an existing apikit configuration file, grabs the element from it, then the element from the incoming
+   * API. It checks if it should update (based on differences between existing and incoming configuration).
+   * Finally applies the change and returns existing configuration.
+   * @param api container of the main mule application file
+   * @param mainMuleConfig main mule configuration (contains http listener, apikit router, console and flows)
+   * @return an existing configuration or null
+   */
+  private MuleConfig updateApikitConfig(ApikitMainFlowContainer api, MuleConfig mainMuleConfig) {
+    MuleConfig apikitConfigMuleConfig = retrieveApikitConfigMuleConfig(mainMuleConfig);
+    Element apikitConfiFromMuleConfig = lookForApikitConfig(apikitConfigMuleConfig);
+    Element apikitConfigFromApi = api.getConfig().generate();
+    MuleConfig apikitConfigResult = null;
     if (shouldUpdateApikitConfig(apikitConfigFromApi, apikitConfiFromMuleConfig)) {
-      int index = config.getContentAsDocument().getRootElement().indexOf(apikitConfiFromMuleConfig);
-      config.getContentAsDocument().getRootElement().removeContent(index);
-      config.getContentAsDocument().getRootElement().addContent(index, apikitConfigFromApi);
+      int index = apikitConfigMuleConfig.getContentAsDocument().getRootElement().indexOf(apikitConfiFromMuleConfig);
+      apikitConfigMuleConfig.getContentAsDocument().getRootElement().removeContent(index);
+      apikitConfigMuleConfig.getContentAsDocument().getRootElement().addContent(index, apikitConfigFromApi);
+      apikitConfigResult = apikitConfigMuleConfig;
     }
+    return apikitConfigResult;
   }
 
-  public MuleConfig createMuleConfig(ApikitMainFlowContainer api) {
-    Document document = new Document();
-    document.setRootElement(new MuleScope(false, false).generate());
-    MuleConfig mConfig = MuleConfigBuilder.fromDoc(document);
-    mConfig.setName(api.getId() + ".xml");
-    setDefaultApikitAndListenersConfigs(api, mConfig);
+  /**
+   * Given a mule configuration, it looks for an element containing `config` for apikit
+    * @param config mule configuration file
+   * @return the element for the apikit configuration
+   */
+  private Element lookForApikitConfig(MuleConfig config) {
```

Comment:
I would call it getChildApikitConfig

Reply:
Adding child but leaving lookFor to differentiate the get methods from the model ones.